src/coingecko/Coingecko.ts

```diff
@@ -72,10 +72,8 @@ export class Coingecko {
     currency = "usd",
     platform_id = "ethereum"
   ): Promise<[string, number]> {
-    const result = await this.getContractDetails(contract_address, platform_id);
```

Comment:
Note that the existing call to getContractDetails retrieved a huge amount of information about the token, only to discard 99% of it.

Using WETH on Ethereum L1 as an example:
https://api.coingecko.com/api/v3/coins/ethereum/contract/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2

https://api.coingecko.com/api/v3/simple/token_price/ethereum?contract_addresses=0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2&vs_currencies=usd

Reply:
Wow amazing improvement - simple yet effective!